modin/backends/pandas/query_compiler.py

```diff
@@ -1601,8 +1585,10 @@ def mode_builder(df, **kwargs):
 
         new_index = pandas.RangeIndex(len(self.index)) if not axis else self.index
         new_columns = self.columns if not axis else pandas.RangeIndex(len(self.columns))
+        new_dtypes = self._get_dtype().copy()
+        new_dtypes.index = new_columns
         return self.__constructor__(
-            new_data, new_index, new_columns, self._dtype_cache
```

Comment:
We use `self._dtype_cache` here to avoid materializing the `dtypes` if they haven't been materialized yet.